web-server/v0.4/src/components/PageHeader/index.js

```diff
@@ -1,23 +1,49 @@
 import React, { PureComponent, createElement } from 'react';
 import PropTypes from 'prop-types';
-import pathToRegexp from 'path-to-regexp';
 import { Breadcrumb, Tabs, Tag } from 'antd';
 import classNames from 'classnames';
 import styles from './index.less';
 import urlToList from '@/components/_utils/pathTools';
 
 const { TabPane } = Tabs;
-export function getBreadcrumb(breadcrumbNameMap, url) {
-  let breadcrumb = breadcrumbNameMap[url];
-  if (!breadcrumb) {
+export const getBreadcrumb = (breadcrumbNameMap, url) => {
+  let breadcrumb = {};
+  if (breadcrumbNameMap[url]) {
+    breadcrumb = breadcrumbNameMap[url];
+  } else {
     Object.keys(breadcrumbNameMap).forEach(item => {
-      if (pathToRegexp(item).test(url)) {
-        breadcrumb = breadcrumbNameMap[item];
+      if (item.includes(url)) {
+        // Returns the route if final path correlates with current path
+        if (
+          item.split('/')[item.split('/').length - 1] === url.split('/')[url.split('/').length - 1]
+        ) {
+          breadcrumb = breadcrumbNameMap[item];
+        }
       }
     });
   }
-  return breadcrumb || {};
-}
+  return breadcrumb;
+};
+
+// Returns the matching path key from breadcrumbNameMap
+// based upon current location
+const getPathKey = (breadcrumbNameMap, url) => {
```

Comment:
Can we refactor this method and the above method to share code? They seem almost identical.

Reply:
+1, we should be able to call `getPathKey()` in `getBreadcrumb()`

Reply:
Absolutely!